proposed/2023/package-source-mapping-actions-PMUI.md

```diff
@@ -0,0 +1,102 @@
+
+ PSM Enabled|Package Mapped|Current Behavior|Proposed Behavior
+---|---|------|------|
+No|No|regular install|regular install
+Yes|No|Error in Error List|Preview Window appears showing mappings being created to selected source
+Yes|Yes|regular install|Allow creating an additional source mapping to selected source for each newly installed package (including transitives)
+ 
+ The existing Preview Window dialog that shows when the customer hits `Install` or `Update` will be modified to list the mappings that will be made by their action. The customer already has the option to disable this preview dialog, so if they choose to disable it, there will be nothing telling them what mappings they made. 
+ 
+ A proposal for the changes to the preview window:
+![PM UI - a proposal for the changes to the preview window](../../meta/resources/PackageSourceMapping/PMUI_preview.png)
+
+#### Restore Errors
+
+There are a few errors the customer could make while trying to install or update a package with a mapping that would make `Restore` fail. First, there could be issues with the transitive packages. Maybe there are transitive packages that are already mapped to a different source or the source that the customer is trying to map a package to does not support some of the transitive packages. 
+
+The customer could also try to update a package that is already mapped to a different source, but the source it is mapped to does not support the version they are trying to update to. In these scenarios, `Restore` will fail in the same way it does today. The customer can see why `Restore` failed in the output window. 
+
+Currently, `NU1100` is used for both an incompatible TFM, and for a missing Package Source Mapping. 
+A proposal to make more specific errors for restore errors related to package source mapping has been made:
+- NU1100 is used for both TFM incompatibility and for an unresolved PackageSourceMapping [#12606](https://github.com/NuGet/Home/issues/12606)
+
+### Technical Explanation
+
+Package Management action logic will pass down a new mapping package ID and source name to Restore. Preview Restore Utilities will look for this new mapping and use it in the preview restore: By appending 2 package patterns, a "*" and the new package ID, along with the new mapping source name to Settings in memory.
+
+Once preview restore determines the newly installed package IDs (top-level and transitive), those IDs will be checked against existing mappings. If an existing mapping is found, no new mapping is written for that package.
+
+If a package is found in the Global Packages Folder (GPF), that package will be mapped to the selected source.
```

Comment:
Can you more specific, what is the selected source in this? The previously downloaded one?

Reply:
Selected source always refer to the upper right drop-down selection in the PM UI. I can add the terms used at the beginning of the spec for clarity.

Reply:
I think terms is a great idea. 

That distinction is important in that scenario since there's a *chance* that the package version on the mapped sources and global packages folder aren't the same.